src/csync/CMakeLists.txt

```diff
@@ -89,52 +59,48 @@ else()
     )
 endif()
 
+if(NOT HAVE_ASPRINTF AND NOT HAVE___MINGW_ASPRINTF)
+    list(APPEND csync_SRCS std/asprintf.c)
+endif()
+
+if (USE_OUR_OWN_SQLITE3)
+    list(APPEND csync_SRCS ${SQLITE3_SOURCE})
+endif()
 
 configure_file(csync_version.h.in ${CMAKE_CURRENT_BINARY_DIR}/csync_version.h)
 
-set(csync_HDRS
-  ${CMAKE_CURRENT_BINARY_DIR}/csync_version.h
-  csync.h
-  vio/csync_vio.h
-  vio/csync_vio_method.h
-  vio/csync_vio_module.h
+set(CSYNC_LIBRARY ocsync)
+add_library(${CSYNC_LIBRARY} SHARED ${common_SOURCES} ${csync_SRCS})
+
+target_include_directories(
+  ${CSYNC_LIBRARY}
+  PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/std
 )
 
-# Statically include sqlite
+find_package(SQLite3 3.8.0 REQUIRED)
 if (USE_OUR_OWN_SQLITE3)
-    list(APPEND csync_SRCS ${SQLITE3_SOURCE})
-    if (WIN32)
-        # We want to export sqlite symbols from the ocsync DLL without
-        # having to patch both sqlite3.h and the amalgation sqlite3.c,
-        # so do the import/export magic manually through the build system.
-        remove_definitions(-DSQLITE_API=__declspec\(dllimport\))
-        add_definitions(-DSQLITE_API=__declspec\(dllexport\))
-    endif()
```

Comment:
Why isn't this needed anymore?

Reply:
Because sqlite is statically built into csync.

Before it was also used from the syncjournaldb, but the dependence was inverted by some refactor before 2.4 and the syncjournaldb is now used by csync.

So now there is only csync that uses sqlite3 symbols, and we don't need this trick to get it work with libowncloudsync.  

This is not because of this patch

Reply:
Okay, thanks for the explanation!